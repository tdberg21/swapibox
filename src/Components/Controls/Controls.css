.controls-container {
  display: flex;
  justify-content: space-evenly
}
.controls-button {
  color: yellow;
  border: 1px solid yellow;
  border-radius: 25px;
  width: 20%;
  height: 100px;
  background-color: #333;
  font-size: 2rem;
}