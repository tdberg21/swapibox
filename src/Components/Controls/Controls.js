import React from 'react';
import PropTypes from 'prop-types';

const Controls = ({ getData, getPlanetData, getVehicleData }) => {

  return (
    <div>
      <button
        onClick={(event) => getData(event)}
        title="people"
        className="people-button"
      >
        People
      </button>
      <button
        onClick={(event) => getPlanetData(event)}
        title="planets"
        className="planet-button"
      >
      Planets
      </button>
      <button
        onClick={(event) => getVehicleData(event)}
        title="vehicles"
        className="vehicle-button"
      >
      Vehicles
      </button>
    </div>
  );
};

Controls.propTypes = {
  getData: PropTypes.func
};

export default Controls;


