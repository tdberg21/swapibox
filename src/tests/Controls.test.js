import React from 'react';
import Controls from '../Components/Controls/Controls.js';
import { shallow, mount } from 'enzyme';


describe('CONTROLS TESTS', () => {
  let mockGetData = jest.fn();
  let mockGetPlanetData = jest.fn();
  let mockVehicleData = jest.fn();
  let wrapper;

  beforeEach(() => {
    wrapper = mount(<Controls getData={mockGetData} getPlanetData={mockGetPlanetData} getVehicleData={mockVehicleData}/>);
  });

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should invoke getData when the people button is clicked', () => {
    wrapper.find('.people-button').simulate('click');

    expect(mockGetData).toHaveBeenCalled();
  });

  it('should invoke getPlanetData when the planet button is clicked', () => {
    wrapper.find('.planet-button').simulate('click');

    expect(mockGetPlanetData).toHaveBeenCalled();
  });

  it('should invoke getVehicleData when the vehicle button is clicked', () => {
    wrapper.find('.vehicle-button').simulate('click');

    expect(mockVehicleData).toHaveBeenCalled();
  });
});

