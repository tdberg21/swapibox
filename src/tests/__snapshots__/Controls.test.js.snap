// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CONTROLS TESTS matches the snapshot 1`] = `
ReactWrapper {
  "length": 1,
  Symbol(enzyme.__unrendered__): <Controls
    getData={[Function]}
    getPlanetData={[Function]}
    getVehicleData={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "getData": [Function],
      "getPlanetData": [Function],
      "getVehicleData": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": <div>
        <button
                title="people"
                class="people-button"
        >
                People
        </button>
        <button
                title="planets"
                class="planet-button"
        >
                Planets
        </button>
        <button
                title="vehicles"
                class="vehicle-button"
        >
                Vehicles
        </button>
</div>,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="people-button"
            onClick={[Function]}
            title="people"
>
            People
</button>,
          <button
            className="planet-button"
            onClick={[Function]}
            title="planets"
>
            Planets
</button>,
          <button
            className="vehicle-button"
            onClick={[Function]}
            title="vehicles"
>
            Vehicles
</button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": <button
            title="people"
            class="people-button"
>
            People
</button>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "People",
            "className": "people-button",
            "onClick": [Function],
            "title": "people",
          },
          "ref": null,
          "rendered": Array [
            "People",
          ],
          "type": "button",
        },
        Object {
          "instance": <button
            title="planets"
            class="planet-button"
>
            Planets
</button>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Planets",
            "className": "planet-button",
            "onClick": [Function],
            "title": "planets",
          },
          "ref": null,
          "rendered": Array [
            "Planets",
          ],
          "type": "button",
        },
        Object {
          "instance": <button
            title="vehicles"
            class="vehicle-button"
>
            Vehicles
</button>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Vehicles",
            "className": "vehicle-button",
            "onClick": [Function],
            "title": "vehicles",
          },
          "ref": null,
          "rendered": Array [
            "Vehicles",
          ],
          "type": "button",
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "getData": [Function],
        "getPlanetData": [Function],
        "getVehicleData": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": <div>
          <button
                    title="people"
                    class="people-button"
          >
                    People
          </button>
          <button
                    title="planets"
                    class="planet-button"
          >
                    Planets
          </button>
          <button
                    title="vehicles"
                    class="vehicle-button"
          >
                    Vehicles
          </button>
</div>,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              className="people-button"
              onClick={[Function]}
              title="people"
>
              People
</button>,
            <button
              className="planet-button"
              onClick={[Function]}
              title="planets"
>
              Planets
</button>,
            <button
              className="vehicle-button"
              onClick={[Function]}
              title="vehicles"
>
              Vehicles
</button>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": <button
              title="people"
              class="people-button"
>
              People
</button>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "People",
              "className": "people-button",
              "onClick": [Function],
              "title": "people",
            },
            "ref": null,
            "rendered": Array [
              "People",
            ],
            "type": "button",
          },
          Object {
            "instance": <button
              title="planets"
              class="planet-button"
>
              Planets
</button>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Planets",
              "className": "planet-button",
              "onClick": [Function],
              "title": "planets",
            },
            "ref": null,
            "rendered": Array [
              "Planets",
            ],
            "type": "button",
          },
          Object {
            "instance": <button
              title="vehicles"
              class="vehicle-button"
>
              Vehicles
</button>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Vehicles",
              "className": "vehicle-button",
              "onClick": [Function],
              "title": "vehicles",
            },
            "ref": null,
            "rendered": Array [
              "Vehicles",
            ],
            "type": "button",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
